File: src/test/java/com/winterhate/strategy/decisions/strategies/DecisionOneStrategy.java
package com.winterhate.strategy.decisions.strategies;

import com.winterhate.strategy.decisions.DecisionStrategy;
import com.winterhate.strategy.decisions.dto.DecisionDto;
import lombok.NonNull;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class DecisionOneStrategy extends DecisionStrategy<DecisionDto> {

    @Override
    public boolean isApplicable(@NonNull DecisionDto dto) {
        return "one".equals(dto.type());
    }

    @Override
    public void apply(DecisionDto dto) {
        log.info("Handled by DecisionOneStrategy: {}", dto);
    }

}

File: src/test/kotlin/com/winterhate/strategy/decisions/strategies/DecisionOneStrategy.kt
package com.winterhate.strategy.decisions.strategies

import com.winterhate.strategy.decisions.DecisionStrategy
import com.winterhate.strategy.decisions.dto.DecisionDto
import org.slf4j.Logger
import org.springframework.stereotype.Component

@Component
class DecisionOneStrategy(
    private val log: Logger
) : DecisionStrategy<DecisionDto> {

    override fun isApplicable(input: DecisionDto): Boolean = input.type == "one"

    override fun apply(data: DecisionDto) {
        log.info("Handled by DecisionOneStrategy: {}", data)
    }

}

